common.go

```diff
@@ -0,0 +1,104 @@
+package common
+
+import (
+	"net/http"
+
+	"github.com/sirupsen/logrus"
+	"github.com/zeebo/errs"
+)
+
+var (
+	ServerError     = errs.Class("Server")
+	ValidationError = errs.Class("Validation")
+	MarshalError    = errs.Class("MarshalJSON")
+	UnmarshalError  = errs.Class("UnmarshalJSON")
+)
+
+func logServerError(err error) {
+	logrus.WithError(err).Error("server error")
+}
+
+func logValidationError(err error) {
+	logrus.WithError(err).Warn("validation error")
+}
+
+func logMarshalError(err error) {
+	logrus.WithError(err).Warn("JSON decoding error")
```

Comment:
marshalling = encoding

also, what are your thoughts on why this kind of error should be treated distinctly from a server error? is there benefit to considering it separately?